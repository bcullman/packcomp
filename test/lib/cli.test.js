const cli = require('../../lib/cli');

describe('cli', () => {
  test('should get dependencies', () => {
    const procArgv = [
      'node',
      'cli.js',
      '-f',
      'n*',
      'test/fixtures/1/',
      'test/fixtures/2/package.json',
      'test/fixtures/3/',
    ];

    global.console = {
      log: (arg) => {
        expect(arg).toMatchSnapshot();
      },
      table: (arg) => {
        expect(arg).toMatchSnapshot();
      },
    };

    cli(procArgv);

    expect.assertions(6);
  });

  test('should get help', () => {
    const procArgv = [
      'node',
      'cli.js',
      'help',
    ];

    global.console = {
      log: jest.fn(),
    };

    cli(procArgv);

    expect(global.console.log).toHaveBeenCalledTimes(1);
  });
});
