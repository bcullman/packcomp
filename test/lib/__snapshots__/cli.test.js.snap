// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli should get dependencies 1`] = `
"
Dependencies:"
`;

exports[`cli should get dependencies 2`] = `
Array [
  Object {
    "Module": "moment",
    "http-status-check": "missing",
    "nth-day": "2.19.4",
    "packcomp": "missing",
  },
  Object {
    "Module": "async",
    "http-status-check": "2.6.1",
    "nth-day": "missing",
    "packcomp": "missing",
  },
  Object {
    "Module": "debug",
    "http-status-check": "3.1.0",
    "nth-day": "missing",
    "packcomp": "3.1.0",
  },
  Object {
    "Module": "lodash",
    "http-status-check": "4.17.10",
    "nth-day": "missing",
    "packcomp": "4.17.10",
  },
  Object {
    "Module": "nodemailer",
    "http-status-check": "4.6.5",
    "nth-day": "missing",
    "packcomp": "missing",
  },
  Object {
    "Module": "request",
    "http-status-check": "2.87.0",
    "nth-day": "missing",
    "packcomp": "missing",
  },
  Object {
    "Module": "string",
    "http-status-check": "3.3.3",
    "nth-day": "missing",
    "packcomp": "missing",
  },
  Object {
    "Module": "sugar",
    "http-status-check": "2.0.4",
    "nth-day": "missing",
    "packcomp": "missing",
  },
  Object {
    "Module": "console.table",
    "http-status-check": "missing",
    "nth-day": "missing",
    "packcomp": "0.10.0",
  },
  Object {
    "Module": "minimist",
    "http-status-check": "missing",
    "nth-day": "missing",
    "packcomp": "1.2.0",
  },
]
`;

exports[`cli should get dependencies 3`] = `
"
DevDependencies:"
`;

exports[`cli should get dependencies 4`] = `
Array [
  Object {
    "Module": "babel-cli",
    "http-status-check": "missing",
    "nth-day": "6.26.0",
    "packcomp": "missing",
  },
  Object {
    "Module": "babel-core",
    "http-status-check": "missing",
    "nth-day": "6.26.0",
    "packcomp": "missing",
  },
  Object {
    "Module": "babel-eslint",
    "http-status-check": "missing",
    "nth-day": "8.0.3",
    "packcomp": "missing",
  },
  Object {
    "Module": "babel-preset-es2015",
    "http-status-check": "missing",
    "nth-day": "6.24.1",
    "packcomp": "missing",
  },
  Object {
    "Module": "babel-preset-stage-1",
    "http-status-check": "missing",
    "nth-day": "6.24.1",
    "packcomp": "missing",
  },
  Object {
    "Module": "babel-register",
    "http-status-check": "missing",
    "nth-day": "6.26.0",
    "packcomp": "missing",
  },
  Object {
    "Module": "eslint",
    "http-status-check": "4.19.1",
    "nth-day": "4.13.1",
    "packcomp": "4.19.1",
  },
  Object {
    "Module": "eslint-config-airbnb-base",
    "http-status-check": "missing",
    "nth-day": "12.1.0",
    "packcomp": "12.1.0",
  },
  Object {
    "Module": "eslint-plugin-security",
    "http-status-check": "missing",
    "nth-day": "1.4.0",
    "packcomp": "missing",
  },
  Object {
    "Module": "mocha",
    "http-status-check": "5.2.0",
    "nth-day": "4.0.1",
    "packcomp": "missing",
  },
  Object {
    "Module": "nyc",
    "http-status-check": "missing",
    "nth-day": "11.3.0",
    "packcomp": "missing",
  },
  Object {
    "Module": "pre-commit",
    "http-status-check": "1.2.2",
    "nth-day": "1.2.2",
    "packcomp": "missing",
  },
  Object {
    "Module": "rimraf",
    "http-status-check": "missing",
    "nth-day": "2.6.2",
    "packcomp": "missing",
  },
  Object {
    "Module": "bunyan",
    "http-status-check": "1.8.12",
    "nth-day": "missing",
    "packcomp": "missing",
  },
  Object {
    "Module": "chai",
    "http-status-check": "4.1.2",
    "nth-day": "missing",
    "packcomp": "missing",
  },
  Object {
    "Module": "coveralls",
    "http-status-check": "3.0.1",
    "nth-day": "missing",
    "packcomp": "missing",
  },
  Object {
    "Module": "eslint-config-guyellis",
    "http-status-check": "0.0.6",
    "nth-day": "missing",
    "packcomp": "missing",
  },
  Object {
    "Module": "istanbul",
    "http-status-check": "0.4.5",
    "nth-day": "missing",
    "packcomp": "missing",
  },
  Object {
    "Module": "npm-update-outdated",
    "http-status-check": "0.1.6",
    "nth-day": "missing",
    "packcomp": "missing",
  },
  Object {
    "Module": "rewire",
    "http-status-check": "4.0.1",
    "nth-day": "missing",
    "packcomp": "missing",
  },
  Object {
    "Module": "eslint-plugin-import",
    "http-status-check": "missing",
    "nth-day": "missing",
    "packcomp": "2.12.0",
  },
  Object {
    "Module": "eslint-plugin-jest",
    "http-status-check": "missing",
    "nth-day": "missing",
    "packcomp": "21.17.0",
  },
  Object {
    "Module": "jest",
    "http-status-check": "missing",
    "nth-day": "missing",
    "packcomp": "23.1.0",
  },
]
`;

exports[`cli should get dependencies 5`] = `
"
peerDependencies:"
`;

exports[`cli should get dependencies 6`] = `Array []`;
